models/correlospinmatrix_detector.cpp

```diff
@@ -0,0 +1,433 @@
+
+      // throw out all binary pulses from event list that are too old to enter the correlation
+      // window
+      BinaryPulselistType& otherPulses = S_.incoming_;
+
+      // calculate the minimum of those neurons that switched on and are not off yet
+      // every impulse in the queue that is further in the past than
+      // this minimum - tau_max cannot contribute to the count covariance
+      long_t t_min_on = t_i_on;
+      for ( int n = 0; n < P_.N_channels_; n++ )
+      {
+        if ( S_.curr_state_[ n ] )
+          if ( S_.last_change_[ n ] < t_min_on )
+            t_min_on = S_.last_change_[ n ];
+      }
+      const double_t tau_edge = P_.tau_max_.get_steps() + P_.delta_tau_.get_steps();
+
+      const delay min_delay = Scheduler::get_min_delay();
+      while (
+        !otherPulses.empty() && ( t_min_on - otherPulses.front().t_off_ ) >= tau_edge + min_delay )
+        otherPulses.pop_front();
+
+
+      // insert new event into history
+      // must happen here so event is taken into account in autocorrelation
+      const BinaryPulse_ bp_i( t_i_on, t_i_off, i );
+
+      BinaryPulselistType::iterator insert_pos = std::find_if( S_.incoming_.begin(),
+        S_.incoming_.end(),
+        std::bind2nd( std::greater< BinaryPulse_ >(), bp_i ) );
+
+      // insert before the position we have found
+      // if no element greater found, insert_pos == end(), so append at the end of the deque
+      S_.incoming_.insert( insert_pos, bp_i );
+
+
+      // go through history of other binary pulses
+      for ( BinaryPulselistType::const_iterator pulse_j = otherPulses.begin();
+            pulse_j != otherPulses.end();
+            ++pulse_j )
+      {
+        // id of other neuron
+        long_t j = pulse_j->receptor_channel_;
+        long_t t_j_on = pulse_j->t_on_;
+        long_t t_j_off = pulse_j->t_off_;
+
+        // minimum and maximum time difference in histogram
+        long_t Delta_ij_min = std::max( t_j_on - t_i_off, -P_.tau_max_.get_steps() );
+        long_t Delta_ij_max = std::min( t_j_off - t_i_on, P_.tau_max_.get_steps() );
+
+        long_t t0 = P_.tau_max_.get_steps() / P_.delta_tau_.get_steps();
+        long_t dt = P_.delta_tau_.get_steps();
+
+
+        // zero time lag covariance
+
+        long_t l = std::min( t_i_off, t_j_off ) - std::max( t_i_on, t_j_on );
+        if ( l > 0 )
+        {
+          S_.count_covariance_[ i ][ j ][ t0 ] += l;
+          if ( i != j )
+            S_.count_covariance_[ j ][ i ][ t0 ] += l;
+        }
+
+        // non-zero time lag covariance
+        for ( long_t Delta = Delta_ij_min / dt; Delta < 0; Delta++ )
+        {
+          long_t l =
+            std::min( t_i_off, t_j_off - Delta * dt ) - std::max( t_i_on, t_j_on - Delta * dt );
+          if ( l > 0 )
+          {
+            S_.count_covariance_[ i ][ j ][ t0 - Delta ] += l;
+            S_.count_covariance_[ j ][ i ][ t0 + Delta ] += l;
+          }
+        }
+
+        if ( i != j )
```

Comment:
Please, always use braces around blocks of code in control structures ([Coding Guidelines](https://nest.github.io/nest-simulator/coding_guidelines_c++#further-indentation-and-formatting)), especially, if they are as nested as here.